Running api with dump of real data (#311)
athenianco/athenian-api: server/README.md

make clean
```

## Running the API server locally with real data

You can also run the api server and all the other services with real data using docker compose. For this setup you'll need the following:
- docker compose,

> - a service account with the corresponding json credentials file to connect to the CloudSQL database,

Reviewer: ```suggestion
- a Google Cloud service account with the corresponding json credentials file to connect to the CloudSQL database,
```

- the credentials for the CloudSQL databases.

Spin up the container with postgres and the container with cloud sql proxy:
```
$ docker-compose up cloud_sql_proxy postgres
```

> This step requires the credentials of the service account as a json file. The default path is `./credentials.json`, otherwise it can be set using the `CLOUD_SQL_PROXY_CREDENTIALS_FILE` env var.

Reviewer: ```suggestion
This step requires the credentials of the service account in a json file. The default path is `./credentials.json`, otherwise it can be set using the `CLOUD_SQL_PROXY_CREDENTIALS_FILE` env var.
```

> Once the dabase is ready to accept connection, run `load_postgre_data.py`. This script accepts a single argument that is the environment name. The accepted values are: `staging` and `production`:

Reviewer: ```suggestion
Once the database is ready to accept connections, run `server/athenian/api/tests/load_postgres_data.py`. This script accepts a single argument that is the environment name. The accepted values are: `staging` and `production`:
```

```
$ CLOUD_SQL_POSTGRES_USER=<name of the pg user> python3 load_postgre_data.py staging
```

This will ask to input the password for the CloudSQL instance as follows:
```
Password for CloudSQL instance of 'staging' environment of user 'staging-cloud-sql':
```

> Then it will dump `state`, `precomputed` and `metadata` databases from the live environment and restores it into the `postgres` container. Here's the example log:

Reviewer: ```suggestion
Then it will dump `state`, `precomputed` and `metadata` databases from the live environment and restores it into the `postgres` container. Here's an example log:
```

Done!
Creating database: createdb -U api state
Loading dump into container: pg_restore -x -Fc -O -U api -d state /db_dumps/state.dump
Done!
Dumping database: pg_dump --host cloud_sql_proxy --port 5432 --username=staging-cloud-sql -Fc --dbname=precomputed > /db_dumps/precomputed.dump
Done!
Creating database: createdb -U api precomputed
Loading dump into container: pg_restore -x -Fc -O -U api -d precomputed /db_dumps/precomputed.dump
Done!
Dumping database: pg_dump --host cloud_sql_proxy --port 5432 --username=staging-cloud-sql -Fc --dbname=metadata > /db_dumps/metadata.dump
Done!
Creating database: createdb -U api metadata
Loading dump into container: pg_restore -x -Fc -O -U api -d metadata /db_dumps/metadata.dump
pg_restore: WARNING:  column "labels" has type "unknown"
DETAIL:  Proceeding with relation creation anyway.
pg_restore: WARNING:  column "milestone_id" has type "unknown"
DETAIL:  Proceeding with relation creation anyway.
pg_restore: WARNING:  column "milestone_title" has type "unknown"
DETAIL:  Proceeding with relation creation anyway.
Done!
```

Now you can finally run the `api`:
```
$ docker-compose up api
```

> By default it runs the latest image, that is `gcr.io/athenian-1/api:latest`. It can be set using the `API_IMAGE` env var.

Reviewer: ```suggestion
By default it runs the latest image, that is `gcr.io/athenian-1/api:latest`. It can be changed using the `API_IMAGE` env var.
```
